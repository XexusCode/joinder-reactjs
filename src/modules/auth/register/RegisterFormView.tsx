import "../login/AuthStyle.scss";
import React from "react";
import { Link } from "react-router-dom";
import { Routes } from "../../../routing/routes";

interface RegisterParams {
  username: string;
  handleInputName: (username: string) => void;
  handleInputPassword: (password: string) => void;
  handleInputPassword2: (password2: string) => void;
  password: string;
  password2: string;
  handleSubmit: (
    e: React.FormEvent<HTMLFormElement>,
    username: string,
    password: string,
    password2: string
  ) => void;
}

export const RegisterFormView = ({
  username,
  password,
  password2,
  handleSubmit,
  handleInputName,
  handleInputPassword,
  handleInputPassword2,
}: RegisterParams) => {
  return (
    <>
      <div className="wrapper fadeInDown">
        <div id="formContent">
          <div className="fadeIn first">
            <img
              src="https://i.pinimg.com/originals/46/bf/f2/46bff2a47eea2421c50e52f1200266d5.png"
              id="icon"
              alt="User Icon"
            />
            <h1>Joinder</h1>
          </div>

          <form
            onSubmit={(e) => {
              e.preventDefault();
              handleSubmit(e, username, password, password2);
            }}
          >
            <input
              type="text"
              id="name"
              className="fadeIn second"
              name="login"
              value={username}
              onChange={(e) => {
                handleInputName(e.target.value);
              }}
              placeholder="Enter your Username"
            />

            <input
              type="password"
              id="password"
              className="fadeIn third"
              name="login"
              value={password}
              onChange={(e) => {
                handleInputPassword(e.target.value);
              }}
              placeholder="Password"
            />

            <input
              type="password"
              id="password2"
              className="fadeIn third"
              name="login"
              value={password2}
              onChange={(e) => {
                handleInputPassword2(e.target.value);
              }}
              placeholder="Repeat Password"
            />
            <input type="submit" className="fadeIn fourth" value="Register" />
          </form>
          <div id="formFooter">
            <Link to={Routes.AUTH_LOGIN}>
              {" "}
              Â¿Ya tienes una cuenta? Logueate{" "}
            </Link>
          </div>
        </div>
      </div>
    </>
  );
};
