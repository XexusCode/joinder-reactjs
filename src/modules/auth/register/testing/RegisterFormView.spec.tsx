import * as React from "react";
import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { Router } from "react-router-dom";
import history from "../../../../routing/history";
import { RegisterFormView } from "../RegisterFormView";

describe("RegiserFormView tests", () => {
  const handleSubmit = jest.fn();
  const handleInputPassword = jest.fn();
  const handleInputName = jest.fn();
  const handleInputPassword2 = jest.fn();

  it("should take a snapshot of <RegisterFormView/>", function () {
    const { asFragment } = render(
      <Router history={history}>
        <RegisterFormView
          handleSubmit={handleSubmit}
          handleInputPassword={handleInputPassword}
          password={"testPassword"}
          username={"testUsername"}
          handleInputName={handleInputName}
          handleInputPassword2={handleInputPassword2}
          password2={"testPassword"}
        />
      </Router>
    );
    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });

  it("the register button calls handleSubmit ", function () {
    const { getByLabelText } = render(
      <Router history={history}>
        <RegisterFormView
          handleSubmit={handleSubmit}
          handleInputPassword={handleInputPassword}
          password={"testPassword"}
          username={"testUsername"}
          handleInputName={handleInputName}
          handleInputPassword2={handleInputPassword2}
          password2={"testPassword"}
        />
      </Router>
    );
    expect(handleSubmit).not.toHaveBeenCalled();
    fireEvent.click(getByLabelText("register-button"));
    expect(handleSubmit).toHaveBeenCalled();
  });
});
