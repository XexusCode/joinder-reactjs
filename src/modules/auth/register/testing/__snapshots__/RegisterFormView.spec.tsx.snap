// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegiserFormView tests should take a snapshot of <RegisterFormView/> 1`] = `
<DocumentFragment>
  <div
    class="wrapper fadeInDown"
  >
    <div
      id="formContent"
    >
      <div
        class="fadeIn first"
      >
        <img
          alt="User Icon"
          id="icon"
          src="https://i.pinimg.com/originals/46/bf/f2/46bff2a47eea2421c50e52f1200266d5.png"
        />
        <h1>
          Joinder
        </h1>
      </div>
      <form>
        <input
          class="fadeIn second"
          id="name"
          name="login"
          placeholder="Enter your Username"
          type="text"
          value="testUsername"
        />
        <input
          class="fadeIn third"
          id="password"
          name="login"
          placeholder="Password"
          type="password"
          value="testPassword"
        />
        <input
          class="fadeIn third"
          id="password2"
          name="login"
          placeholder="Repeat Password"
          type="password"
          value="testPassword"
        />
        <input
          class="fadeIn fourth"
          type="submit"
          value="Register"
        />
      </form>
      <div
        id="formFooter"
      >
        <a
          href="/auth/login"
        >
           Â¿Ya tienes una cuenta? Logueate 
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
